src/hqrrp.cpp

```diff
@@ -703,14 +714,27 @@ int64_t hqrrp( int64_t m_A, int64_t n_A, double * buff_A, int64_t ldim_A,
           buff_pB, buff_sB,
           1, m_A, buff_AR, ldim_A,
           1, m_Y, buff_YR, ldim_Y,
-          0, buff_Y, ldim_Y );
+          0, nullptr, 0 
+      );
     }
 
     //
     // Compute QRP of panel AB1 = [ A11; A21 ].
     // Apply same permutations to A01 and Y1, and build T1_T.
     //
-
+    //    Notes
+    //    -----
+    //    The function below is basically running GEQP3 *or* GEQRF on
+    //    the updated sketch *and then* changing the representation of the
+    //    composition of Householder reflectors.
+    //
+    //    In the code path where we hit a GEQP3-like function we can't use
+    //    GEQP3 directlybecause we actually need to modify three matrices
+    //    (AB1, A01, and Y1) alongside one another.
+    //    
+    //    The code path where we hit a GEQRF-like function is very different;
+    //    it only operates on AB1!
+    //
```

Comment:
@TeachRaccooon this is where HQRRP processes A[:, panel]. I've introduced a new code path that shows when panel_pivoting is off, this basically reduces to GEQRF on AB1 followed by LARFT on the Householder vectors returned by GEQRF.